core/src/HIP/Kokkos_HIP_Locks.cpp

```diff
@@ -65,10 +64,11 @@ void initialize_host_hip_lock_arrays() {
       sizeof(std::int32_t) * (KOKKOS_IMPL_HIP_SPACE_ATOMIC_MASK + 1)));
 
   g_host_hip_lock_arrays.n = HIPInternal::concurrency();
-
-  KOKKOS_COPY_HIP_LOCK_ARRAYS_TO_DEVICE();
+  copy_hip_lock_arrays_to_device();
   init_lock_array_kernel_atomic<<<
       (KOKKOS_IMPL_HIP_SPACE_ATOMIC_MASK + 1 + 255) / 256, 256, 0, nullptr>>>();
+  HIP::impl_static_fence(
+      "Kokkos::Impl::initialize_host_hip_lock_arrays: Post Init Lock Arrays");
```

Comment:
This fence is new (and we have it in the `CUDA` implementation).

Reply:
I don't see why this is necessary for the HIP backend

Reply:
In my opinion, it makes sense to make sure that the kernel finishes before returning so that other streams see the correct values. Noe that we do the same in `desul`: https://github.com/desul/desul/blob/0a0dfd3a4b53030913f970f0a28714a0415e147a/atomics/src/Lock_Array_HIP.cpp#L78-L79
Hopefully, we can get rid of these variables soon anyway and just use the `desul` ones.

Reply:
This has nothing to do with streams. The function is only called when initializing the backend and it is followed by a synchronized HIP function. This just adds pointless code.

Reply:
Not worth fighting over. I removed the fence.